### PR TITLE
Last selected target prioritized in VS Code

### LINKED ISSUE #1348: VSCode

### PR DESCRIPTION
**Pull request by Razz4780**
Resolves #1348

The first entry in target selection list is now the last used target. The rest of the list is sorted.

### CODE REVIEW

File: vscode-ext/src/extension.ts [hunk 1]
```diff
@@ -153,7 +157,23 @@ class MirrordAPI {
 			throw new Error("error occured listing targets: " + stderr);
 		}
 
-		return JSON.parse(stdout);
+		const targets: string[] = JSON.parse(stdout);
+		const lastTarget = globalContext.globalState.get(LAST_TARGET_KEY);
```
Comment (aviramha) [inline vscode-ext/src/extension.ts:161]: I would first check if there's workspace state then use global state.

File: vscode-ext/src/extension.ts [hunk 2]
```diff
@@ -153,7 +157,23 @@ class MirrordAPI {
...
+		targets.sort((t1, t2) => {
```
Comment (aviramha) [inline vscode-ext/src/extension.ts:162]: I would just call `sort` then 
```js
let pop = targets.pop(lastTarget)
if (pop) {
targets.unshift(pop)
}
```

**Review by aviramha — APPROVED** [review]

**Review by aviramha — CHANGES_REQUESTED** [review]
Missclick

File: vscode-ext/src/extension.ts (hunk 2 above)
Comment (Razz4780) [reply vscode-ext/src/extension.ts]: Thanks, indeed looks better

File: vscode-ext/src/extension.ts (hunk 1 above)
Comment (Razz4780) [reply vscode-ext/src/extension.ts]: Added :heavy_check_mark:

**Review by aviramha — APPROVED** [review]
GJ!

**Pull request merged**